skyportal/handlers/api/observation_plan.py

```diff
@@ -1463,45 +1468,50 @@ def observation_simsurvey(
-            The survey efficiency analysis for the request
-        width : float
-            Width of the telescope field of view in degrees.
-        height : float
-            Height of the telescope field of view in degrees.
-        number_of_injections: int
-            Number of simulations to evaluate efficiency with. Defaults to 1000.
-        number_of_detections: int
-            Number of detections required for detection. Defaults to 1.
-        detection_threshold: int
-            Threshold (in sigmas) required for detection. Defaults to 5.
-        minimum_phase: int
-            Minimum phase (in days) post event time to consider detections. Defaults to 0.
-        maximum_phase: int
-            Maximum phase (in days) post event time to consider detections. Defaults to 3.
-        injection_filename: str
-            Path to file for injestion as a simsurvey.models.AngularTimeSeriesSource
-    . Defaults to nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt from https://github.com/mbulla/kilonova_models.
+    Parameters
+    ----------
+    observations : skyportal.models.observation_plan.PlannedObservation
+        The planned observations associated with the request
+    localization_id : int
+        The id of the skyportal.models.localization.Localization that the request is made based on
+    instrument_id : int
+        The id of the skyportal.models.instrument.Instrument that the request is made based on
+    survey_efficiency_analysis_id : int
+        The id of the survey efficiency analysis for the request (either skyportal.models.survey_efficiency.SurveyEfficiencyForObservations or skyportal.models.survey_efficiency.SurveyEfficiencyForObservationPlan).
+    survey_efficiency_analysis_type : str
+        Either SurveyEfficiencyForObservations or SurveyEfficiencyForObservationPlan.
+    width : float
+        Width of the telescope field of view in degrees.
+    height : float
+        Height of the telescope field of view in degrees.
+    number_of_injections : int
+        Number of simulations to evaluate efficiency with. Defaults to 1000.
+    number_of_detections : int
+        Number of detections required for detection. Defaults to 1.
+    detection_threshold : int
+        Threshold (in sigmas) required for detection. Defaults to 5.
+    minimum_phase : int
+        Minimum phase (in days) post event time to consider detections. Defaults to 0.
+    maximum_phase : int
+        Maximum phase (in days) post event time to consider detections. Defaults to 3.
+    model_name : str
+        Model to simulate efficiency for. Must be one of kilonova, afterglow, or linear. Defaults to kilonova.
+    optional_injection_parameters: dict
+        Optional parameters to specify the injection type, along with a list of possible values (to be used in a dropdown UI)
     """
 
     session = Session()
```

Comment:
I'm having a hard time because there are a lot of functions calling other functions, but I noticed some of the "simsurvey" functions get `session` as an argument and some don't (and open a session internally). Are you sure this is consistent behavior that makes sense? My first guess is that the session should be made in the handler and passed into the functions, but you may have other considerations...